### ISSUE TITLE
Shop NPCs with Conditions seem to be broke in this new update

### ISSUE THREAD

**Issue by PandahutMiku** [SmartlyDressedGames/Unturned-3.x-Community#1908]
I tested this on Unturnov Vanilla Unturned Server, zero Rocketmod or openmod. Trying to sell something to a vendor that has a condition just fails and throws this error in console. Doesn't seem to happen in singleplayer.

```[2020-07-31 20:10:17] Exception raised when RPC invoked askSellToVendor:
[2020-07-31 20:10:17] Exception has been thrown by the target of an invocation.
[2020-07-31 20:10:17]   at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00048] in <e1319b7195c343e79b385cd3aa43f5dc>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <e1319b7195c343e79b385cd3aa43f5dc>:0 
  at SDG.Unturned.SteamChannel.receive (Steamworks.CSteamID steamID, System.Byte[] packet, System.Int32 offset, System.Int32 size) [0x003bb] in <4905e585b2bf44069c6b6bf385202de7>:0 
[2020-07-31 20:10:17] Object reference not set to an instance of an object
[2020-07-31 20:10:17]   at SDG.Unturned.NPCShortFlagCondition.isConditionMet (SDG.Unturned.Player player) [0x00000] in <4905e585b2bf44069c6b6bf385202de7>:0 
  at SDG.Unturned.DialogueElement.areConditionsMet (SDG.Unturned.Player player) [0x00012] in <4905e585b2bf44069c6b6bf385202de7>:0 
  at SDG.Unturned.DialogueAsset.getAvailableResponses (System.Int32 messageIndex, System.Collections.Generic.List`1[T] availableResponses) [0x000ab] in <4905e585b2bf44069c6b6bf385202de7>:0 
  at SDG.Unturned.DialogueAsset.doesPlayerHaveAccessToVendorInternal (SDG.Unturned.Player player, SDG.Unturned.VendorAsset vendorAsset, System.Int32 depth) [0x0001c] in <4905e585b2bf44069c6b6bf385202de7>:0 
  at SDG.Unturned.DialogueAsset.doesPlayerHaveAccessToVendor (SDG.Unturned.Player player, SDG.Unturned.VendorAsset vendorAsset) [0x00000] in <4905e585b2bf44069c6b6bf385202de7>:0 
  at SDG.Unturned.ObjectNPCAsset.doesPlayerHaveAccessToVendor (SDG.Unturned.Player player, SDG.Unturned.VendorAsset vendorAsset) [0x00016] in <4905e585b2bf44069c6b6bf385202de7>:0 
  at SDG.Unturned.PlayerQuests.askSellToVendor (Steamworks.CSteamID steamID, System.UInt16 id, System.Byte index) [0x000b9] in <4905e585b2bf44069c6b6bf385202de7>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke(System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00032] in <e1319b7195c343e79b385cd3aa43f5dc>:0 ```

**Comment by FANYPLAYRU** [comment]
Yes, i have this problem

**Comment by PandahutMiku** [comment]
[Server_Pandahut_#19_[Super-Duper-Extra-Secret-TestServer].log](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/files/5008417/Server_Pandahut_.19_.Super-Duper-Extra-Secret-TestServer.log)

Here's full log if needed. I edited my post to include this now, but just to make sure it is noted, this seems to not happen in singleplayer at all, only in multiplayer.

**Comment by SDGNelson** [comment]
Sorry about this! I am not sure how I didn't run into it while testing. Will fix in a patch this afternoon, but keeping an eye out for any other issues to include in the patch as well.

**Comment by PandahutMiku** [comment]
> Sorry about this! I am not sure how I didn't run into it while testing. Will fix in a patch this afternoon, but keeping an eye out for any other issues to include in the patch as well.

It only happens in multiplayer and only effects NPCs with have conditional logic to them it seems. Like how Unturnov will let you level up your NPCs to gain access to higher shops, and it uses conditional logic to show those shops/determine if you have access to them.

Elver NPC's are uneffected, as well as any normal NPC like Scrap Mechanic on Unturnov which is just a simple shop.

**Comment by SDGNelson** [comment]
This should be resolved in the 3.20.8.1 patch: https://steamcommunity.com/app/304930/discussions/0/2785991898397739840/
